pymc/tests/sampling/test_mcmc.py

```diff
@@ -819,6 +820,22 @@ def test_modify_step_methods(self):
                 steps = assign_step_methods(model, [])
         assert isinstance(steps, NUTS)
 
+    def test_step_vars_in_model(self):
+        """Test if error is raised if step variable is not found in model.value_vars"""
+        with pm.Model() as model:
+            c1 = pm.HalfNormal("c1")
+            c2 = pm.HalfNormal("c2")
+
+            with pytensor.config.change_flags(mode=fast_unstable_sampling_mode):
+                step1 = NUTS([c1])
+                step2 = NUTS([c2])
+                step2.vars = [
+                    at.dscalar("x"),
+                ]
```

Comment:
I think this is a pretty good way to test the functionality. I would suggest specifically passing the random variable `c2` as that was the mistake the user did in the original issue.

```suggestion
                step2.vars = [c2]
```

Reply:
done!